envoy/config/filter/http/jwt_authn/v2/config.proto

```diff
@@ -0,0 +1,260 @@
+//     jwks_uri: https://example.com/.well-known/jwks.json
+//
+// [#not-implemented-hide:]
+message JWT {
+  // Identifies the principal that issued the JWT. See
+  // `here <https://tools.ietf.org/html/rfc7519#section-4.1.1>`_.
+  // Usually a URL or an email address.
+  //
+  // Example: https://securetoken.google.com
+  // Example: 1234567-compute@developer.gserviceaccount.com
+  //
+  string issuer = 1;
+
+  // The list of JWT `audiences
+  // <https://tools.ietf.org/html/rfc7519#section-4.1.3>`_. that are allowed to
+  // access. A JWT containing any of these audiences will be accepted.
+  //
+  // Example:
+  //
+  // .. code-block:: yaml
+  //
+  //     audiences:
+  //     - bookstore_android.apps.googleusercontent.com
+  //       bookstore_web.apps.googleusercontent.com
+  //
+  repeated string audiences = 2;
+
+  // `Json web key set <https://tools.ietf.org/html/rfc7517#appendix-A>`_ URI.
+  // Json web key set is fetched from the URI to validate signature of the JWT.
+  // See `OpenID Discovery
+  // <https://openid.net/specs/openid-connect-discovery-1_0.html#ProviderMetadata>`_
+  //
+  // Example:  jwks_uri:
+  //
+  // .. code-block:: yaml
+  //
+  //    jwks_uri: https://www.googleapis.com/oauth2/v1/certs
+  //
+  string jwks_uri = 3;
+
+  // This field is specific for Envoy proxy implementation.
+  // It is the cluster name in the Envoy "cluster_manager" config section.
+  // In order for Envoy to call "jwks_uri", its host has to be specified
+  // as a "cluster" in the config for each jwks_uri.
+  //
+  // Example:  jwks cluster in "cluster_manager" section of Envoy config.
+  //
+  // .. code-block:: json
+  //
+  //    {
+  //      "name": "jwt.www.googleapis.com|443",
+  //      "connect_timeout_ms": 1000,
+  //      "type": "strict_dns",
+  //      "lb_type": "round_robin",
+  //         "hosts": [
+  //            {
+  //              "url": "tcp://www.googleapis.com:443"
+  //            }
+  //          ],
+  //      "ssl_context": {},
+  //      "circuit_breakers": {
+  //         "default": {
+  //           "max_pending_requests": 10000,
+  //           "max_requests": 10000
+  //         }
+  //      }
+  //    }
+  //
+  // Example:  jwks_uri_cluster:
+  //
+  // .. code-block:: yaml
+  //
+  //    jwks_uri: https://www.googleapis.com/oauth2/v1/certs
+  //    jwks_uri_cluster: jwt.www.googleapis.com|443
+  //
+  string jwks_uri_cluster = 4;
+
+  // Duration after which the cached public key should be expired. If not
+  // specified, default cache duration is 5 minutes.
+  google.protobuf.Duration public_key_cache_duration = 5;
+
+  // If false, the JWT is removed in the request after a success verification.
+  // Set it true, if don't want it to be removed.
+  // Default value is false which is to remove JWT.
+  bool forward_jwt = 6;
+
+  // If true, the request is allowed if the JWT verification fails.
+  // Default value is false; JWT verification failed request will be rejected.
```

Comment:
This is a good place to clarify the failure mode. Is rejection here an HTTP unauthorized error code?